src/main/java/cubicchunks/worldgen/gui/FlatLayersTab.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.worldgen.gui;
+
+import static cubicchunks.worldgen.gui.CustomCubicGui.HORIZONTAL_INSETS;
+import static cubicchunks.worldgen.gui.CustomCubicGui.HORIZONTAL_PADDING;
+import static cubicchunks.worldgen.gui.CustomCubicGui.VERTICAL_INSETS;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import cubicchunks.worldgen.generator.flat.FlatGeneratorSettings;
+import cubicchunks.worldgen.generator.flat.Layer;
+import java.util.Comparator;
+import cubicchunks.worldgen.gui.component.UIFlatTerrainLayer;
+import cubicchunks.worldgen.gui.component.UIVerticalTableLayout;
+import net.malisis.core.client.gui.component.UIComponent;
+import net.malisis.core.client.gui.component.container.UIContainer;
+
+public class FlatLayersTab implements Comparator<UIFlatTerrainLayer> {
```

Comment:
This could be refactored into some kind of UIComponent, but this should be ok for now